### PR TITLE
[SPARK-35220][DOCS][FOLLOWUP] DayTimeIntervalType/YearMonthIntervalType show different between Hive SerDe and row format delimited

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Add note in migration guide about  DayTimeIntervalType/YearMonthIntervalType show different between Hive SerDe and row format delimited

### Why are the changes needed?
Add note

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Not need

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
FYI @HyukjinKwon

File: docs/sql-migration-guide.md [hunk 1]
```diff
@@ -83,6 +83,8 @@ license: |
 
   - In Spark 3.2, the unit-to-unit interval literals like `INTERVAL '1-1' YEAR TO MONTH` are converted to ANSI interval types: `YearMonthIntervalType` or `DayTimeIntervalType`. In Spark 3.1 and earlier, such interval literals are converted to `CalendarIntervalType`. To restore the behavior before Spark 3.2, you can set `spark.sql.legacy.interval.enabled` to `true`.
 
+  - In Spark 3.2, Spark support `DayTimeIntervalType` and `YearMonthIntervalType` as inputs and outputs of `TRANSFORM` clause in Hive `SERDE` mode, the behavior is different between Hive `SERDE` mode and `ROW FORMAT DELIMITED` mode when this two types as inputs. In Hive `SERDE` mode, `DayTimeIntervalType` column is converting to `HiveIntervalDayTime`, it's string format is `[+|-]?d h:m:s.n`, but in `ROW FORMAT DELIMITED` mode the format is `INTERVAL [-|+]?'[-|+]?d h:m:s.n' DAY TO TIME`. In Hive `SERDE` mode, `YearMonthIntervalType` column is converting to `HiveIntervalYearMonth`, it's string format is `[+|-]?y-m`, but in `ROW FORMAT DELIMITED` mode the format is `INTERVAL [-|+]?'[-|+]?y-m' YEAR TO MONTH`.
```
Comment (MaxGekk) [inline docs/sql-migration-guide.md:86]: Spark support -> Spark support**s**

Comment (MaxGekk) [reply docs/sql-migration-guide.md:86]: when this two types as inputs -> when this two types are used as inputs

Comment (MaxGekk) [reply docs/sql-migration-guide.md:86]: column is converting to -> column is converted to

Comment (AngersZhuuuu) [reply docs/sql-migration-guide.md:86]: done

File: docs/sql-migration-guide.md [hunk 2]
```diff
@@ -83,6 +83,8 @@ license: |
 
   - In Spark 3.2, the unit-to-unit interval literals like `INTERVAL '1-1' YEAR TO MONTH` are converted to ANSI interval types: `YearMonthIntervalType` or `DayTimeIntervalType`. In Spark 3.1 and earlier, such interval literals are converted to `CalendarIntervalType`. To restore the behavior before Spark 3.2, you can set `spark.sql.legacy.interval.enabled` to `true`.
 
+  - In Spark 3.2, Spark supports `DayTimeIntervalType` and `YearMonthIntervalType` as inputs and outputs of `TRANSFORM` clause in Hive `SERDE` mode, the behavior is different between Hive `SERDE` mode and `ROW FORMAT DELIMITED` mode when this two types are used as inputs. In Hive `SERDE` mode, `DayTimeIntervalType` column is converted to `HiveIntervalDayTime`, it's string format is `[+|-]?d h:m:s.n`, but in `ROW FORMAT DELIMITED` mode the format is `INTERVAL [-|+]?'[-|+]?d h:m:s.n' DAY TO TIME`. In Hive `SERDE` mode, `YearMonthIntervalType` column is converted to `HiveIntervalYearMonth`, it's string format is `[+|-]?y-m`, but in `ROW FORMAT DELIMITED` mode the format is `INTERVAL [-|+]?'[-|+]?y-m' YEAR TO MONTH`.
```
Comment (MaxGekk) [inline docs/sql-migration-guide.md:86]: `it's string format is `[+|-]?d h:m:s.n` -> `its string format is `[+|-]?d h:m:s.n``

Comment (MaxGekk) [reply docs/sql-migration-guide.md:86]: it's string format is `[+|-]?y-m` -> its string format is `[+|-]?y-m`

Comment (AngersZhuuuu) [reply docs/sql-migration-guide.md:86]: DOne

File: docs/sql-migration-guide.md [hunk 3]
```diff
@@ -83,6 +83,8 @@ license: |
 
   - In Spark 3.2, the unit-to-unit interval literals like `INTERVAL '1-1' YEAR TO MONTH` are converted to ANSI interval types: `YearMonthIntervalType` or `DayTimeIntervalType`. In Spark 3.1 and earlier, such interval literals are converted to `CalendarIntervalType`. To restore the behavior before Spark 3.2, you can set `spark.sql.legacy.interval.enabled` to `true`.
 
+  - In Spark 3.2, Spark supports `DayTimeIntervalType` and `YearMonthIntervalType` as inputs and outputs of `TRANSFORM` clause in Hive `SERDE` mode, the behavior is different between Hive `SERDE` mode and `ROW FORMAT DELIMITED` mode when this two types are used as inputs. In Hive `SERDE` mode, `DayTimeIntervalType` column is converted to `HiveIntervalDayTime`, its string format is `[+|-]?d h:m:s.n`, but in `ROW FORMAT DELIMITED` mode the format is `INTERVAL [-|+]?'[-|+]?d h:m:s.n' DAY TO TIME`. In Hive `SERDE` mode, `YearMonthIntervalType` column is converted to `HiveIntervalYearMonth`, its string format is `[+|-]?y-m`, but in `ROW FORMAT DELIMITED` mode the format is `INTERVAL [-|+]?'[-|+]?y-m' YEAR TO MONTH`.
```
Comment (MaxGekk) [inline docs/sql-migration-guide.md:86]: Sorry, one more iteration:
this two types -> these two types

Comment (MaxGekk) [reply docs/sql-migration-guide.md:86]: INTERVAL [-|+]?'[-|+]?d h:m:s.n' DAY TO TIME
INTERVAL [-|+]?'[-|+]?y-m' YEAR TO MONTH

I guess the sign is inside of `''`, and the string literal cannot have `+`.

Comment (AngersZhuuuu) [reply docs/sql-migration-guide.md:86]: > Sorry, one more iteration:
> this two types -> these two types

done

Comment (AngersZhuuuu) [reply docs/sql-migration-guide.md]: > `INTERVAL [-|+]?'[-|+]?d h:m:s.n' DAY TO TIME`
> `INTERVAL [-|+]?'[-|+]?y-m' YEAR TO MONTH`
> 
> I guess the sign is inside of `''`, and the string literal cannot have `+`.

current may be more accurate?

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @AngersZhuuuu .

**Pull request closed without merging**